File: Doubly LL/C++/arr2ll.cpp
#include <iostream>
using namespace std;
class Node {
    public:
    int data;
    Node* next;
    Node* prev;
    
    public:
    Node(int data1, Node* next1, Node* prev1) {
        this->data = data1;
        this->next = next1;
        this->prev = prev1;
    }
    
    public:
    Node(int data1) {
        this->data = data1;
        this->next = nullptr;
        this->prev = nullptr;
    }
};
Node* arrayToDLL(int arr[], int n) {
    Node* head = new Node(arr[0]);
    Node* prev = head;
    for (int i=1; i<n; i++) {
        Node* temp = new Node(arr[i]);
        prev->next = temp;
        prev = temp;
    }
    return head;
}
void print(Node* head) {

    Node* temp = head;

    while (temp != NULL) {
        cout << temp->data << " ";
        temp = temp->next;
    }

    cout << endl;
}

int main() {

    int arr[] = {10, 20, 30, 40, 50};
    int n = sizeof(arr) / sizeof(arr[0]);

    Node* head = arrayToDLL(arr, n);

    print(head);

    return 0;
}

File: Doubly LL/Java/arr2ll.java
class Node {
    int data;
    Node next;
    Node prev;

    Node(int data, Node next, Node prev) {
        this.data = data;
        this.next = next;
        this.prev = prev;
    }
    Node(int data) {
        this.data = data;
        this.next = null;
        this.prev = null;
    }
};
public class arr2ll {

    static Node convertArr2DLL(int[] arr, int n) {
        if (n == 0) {
            return null;
        }
        Node head = new Node(arr[0]);
        Node prev = head;
        for (int i = 1; i < n; i++) {
            Node temp = new Node(arr[i]);
            prev.next = temp;
            // temp.prev = prev;
            // prev = prev.next;
            prev = temp;
        }
        return head;
    }

    static void print(Node head) {
        Node prev = head;
        while (prev != null) {
            System.out.print(prev.data + " ");
            prev = prev.next;
        }
        System.out.println();
    }
    public static void main(String[] args) {
        int[] arr = {10, 20, 30, 40, 50};
        int n = arr.length;
        Node head = convertArr2DLL(arr, n);
        print(head);
    }
}